### PR TITLE
fix: [IOBP-128] Fix bottom sheet padding regression

### PR DESCRIPTION
**Pull request by mastro993**
## Short description
This PR fixes a regression in the bottom sheet modal visualization

## List of changes proposed in this pull request
- `ts/utils/hooks/bottomSheet.tsx`: fixed bottom sheet bottom padding regression
- `ts/features/design-system/core/DSBottomSheet.tsx`: added a new type of bottom sheet to test. This bottom sheet has a very long content and its snap point should snap below the upper safe-area limit
- `ts/features/idpay/initiative/timeline/components/TimelineDetailsBottomSheet.tsx`: replaced legacy bottom sheet

## How to test
Navigate to the Bottom Sheet screen in the DS section and make sure all types of modals are displayed correctly.
This fix should be tested on both iOS and Android.

## Acknowledgement

@SabPicariello

### CODE REVIEW

File: ts/features/idpay/payment/components/Barcode/useIOBarcodeFileReader.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import * as A from "fp-ts/lib/Array";
 import * as E from "fp-ts/lib/Either";
 import * as O from "fp-ts/lib/Option";
-import * as TE from "fp-ts/lib/TaskEither";
 import * as T from "fp-ts/lib/Task";
+import * as TE from "fp-ts/lib/TaskEither";
```
Comment (shadowsheep1) [inline ts/features/idpay/payment/components/Barcode/useIOBarcodeFileReader.tsx:5]: I would revert this change...  what do you think? For the sake of "changes minimization principle" 😉

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
Just a little change ;)

**Review by shadowsheep1 — APPROVED** [review]
LGTM

File: ts/features/idpay/payment/components/Barcode/useIOBarcodeFileReader.tsx (hunk 1 above)
Comment (mastro993) [reply ts/features/idpay/payment/components/Barcode/useIOBarcodeFileReader.tsx]: Reverted in https://github.com/pagopa/io-app/pull/4784/commits/06eee14b2d05d343daa3ae5bde40223044e118e5 👌🏻
